Templates/Android/generated/BaseEntityCollectionReferenceRequest.java.tt

```diff
@@ -39,37 +39,43 @@
                 .post(new<#=TypeName(c)#>, body);
     }
 
+<# if (c.GetFeatures().CanExpand) { #>
     /**
      * Sets the expand clause for the request
      *
      * @param value The expand clause
      * @return The updated request
      */
     public <#=ITypeCollectionReferenceRequest(c)#> expand(final String value) {
-        addQueryOption(new QueryOption("expand", value));
+        addQueryOption(new QueryOption("$expand", value));
         return (<#=TypeCollectionReferenceRequest(c)#>)this;
     }
 
+<# } #>
+<# if (c.GetFeatures().CanSelect) { #>
```

Comment:
RE: using template logic, I'd recommend a new template file that can be imported into all the requests for the odata clauses.

---
In reply to: [69333050](https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/pull/29#discussion_r69333050) [](ancestors = 69333050)